### PR TITLE
adding auth tests

### PR DESCRIPTION
**Pull request by zhaque44**
#### Proposed Changes ####
Writing tests for the auth pkg: this will result in better test coverage

#### Types of Changes ####
This does not require changes to the docs

#### Testing ####

Testing the auth pkg
#### User-Facing Change ####

```release-note

```

### CODE REVIEW

**Comment by zhaque44** [comment]
@dereknola a little help here if you don't mind, my goal is too add tests for the auth pkg:

```
 make test  
./scripts/unit-tests
+ BUILDTAGS='selinux netgo osusergo no_stage static_build sqlite_omit_load_extension no_embedded_executor no_cri_dockerd'
+ GO_BUILDTAGS=' selinux netgo osusergo no_stage static_build sqlite_omit_load_extension no_embedded_executor no_cri_dockerd '
+ go test -tags ' selinux netgo osusergo no_stage static_build sqlite_omit_load_extension no_embedded_executor no_cri_dockerd ' -v -cover ./pkg/... -run Unit
package github.com/rancher/rke2/pkg/cli/cmds
	imports github.com/rancher/rke2/pkg/rke2
	imports github.com/k3s-io/k3s/pkg/cli/agent
	imports github.com/k3s-io/k3s/pkg/agent
	imports github.com/k3s-io/k3s/pkg/cgroups: build constraints exclude all Go files in /Users/zubairhaque/go/pkg/mod/github.com/k3s-io/k3s@v1.29.2-rc1.0.20240221202119-fae0d998631e/pkg/cgroups
package github.com/rancher/rke2/pkg/cli/cmds
	imports github.com/rancher/rke2/pkg/windows: build constraints exclude all Go files in /Users/zubairhaque/Desktop/rke2/pkg/windows
make: *** [unit-tests] Error 1
```
im getting this error, and i cannot seem to run this one test to make sure it works. Any help is appreciated thank

File: go.mod [hunk 1]
```diff
@@ -128,6 +128,8 @@ require (
 	sigs.k8s.io/yaml v1.4.0
 )
 
+require github.com/stretchr/testify v1.8.4
```
Comment (dereknola) [inline go.mod:131]:
```suggestion
```

We don't use testify, we utilize the builting go test generation provided by https://github.com/cweill/gotests which relies on `reflect.DeepEqual` usually. 
See https://github.com/rancher/rke2/blob/master/pkg/rke2/serviceaccount_test.go#L127 and https://github.com/rancher/rke2/blob/master/pkg/rke2/rke2_linux_test.go#L138-L142

**Review by dereknola — CHANGES_REQUESTED** [review]

**Comment by dereknola** [comment]
If you building this on a Mac or Window your likely going to run into issue with go build tags

**Comment by dereknola** [comment]
Try using `make in-docker-test` which wraps the script in the docker build environment.

**Comment by zhaque44** [comment]
> Try using `make in-docker-unit-tests` which wraps the script in the docker build environment.

thanks! so much for all of your feedback, should i turn it into a draft until i can check that the test im writing works? and refactor it to use the right test assertion library?

**Pull request closed without merging**